File: 8kyu/(8 kyu) Returning Strings/(8 kyu) Returning Strings.cpp
// 1 Plain solution
#include <string>

std::string greet(const std::string& n){
  std::string str1 = "Hello, ";
  std::string str2 = " how are you doing today?";

  return str1 + n + str2;
}

// 2 Optimized solution
#include <string>

std::string greet(const std::string& n){
  return "Hello, " + n + " how are you doing today?";
}

// 3 Clever solution
#include <fmt/format.h>

std::string greet( const std::string& name ) {
  return fmt::format( "Hello, {} how are you doing today?", name );
}

File: 8kyu/(8 kyu) Returning Strings/(8 kyu) Returning Strings.rs
// 1 Plain solution
fn greet(name: &str) -> String {
    let str1 = "Hello, ";
    let str2 = " how are you doing today?";

    return format!("{}{}{}", str1, name, str2);
}

// 2 Optimized solution
fn greet_optimized(name: &str) -> String {
    format!("Hello, {} how are you doing today?", name)
}

// 3 Clever solution
fn greet_clever(name: &str) -> String {
    ["Hello, ", name, " how are you doing today?"].concat()
}